pkg/configuration/configuration_test.go

```diff
@@ -439,3 +441,95 @@ func Test_Configuration_GetKeyType(t *testing.T) {
 	assert.Equal(t, EnvVarKeyType, config.GetKeyType("snyk_something"))
 	assert.Equal(t, UnspecifiedKeyType, config.GetKeyType("app"))
 }
+
+func Test_JsonStorage_Locking(t *testing.T) {
+	outerConfig := NewFromFiles(TEST_FILENAME)
+	outerConfig.PersistInStorage("n")
+	outerConfig.Set("n", float64(0))
+	N := 100
+	ch := make(chan struct{}, N)
+	ctx := context.Background()
+	for range N {
+		go func() {
+			defer func() { ch <- struct{}{} }()
+
+			config := NewFromFiles(TEST_FILENAME)
+			config.PersistInStorage("n")
+			storage := config.GetStorage()
+			assert.NotNil(t, storage)
+
+			err := storage.Lock(ctx, time.Millisecond)
+			assert.NoError(t, err)
+			defer func() {
+				unlockErr := storage.Unlock()
+				assert.NoError(t, unlockErr)
+			}()
+
+			err = storage.Refresh(config, "n")
+			assert.NoError(t, err)
+			config.Set("n", config.Get("n").(float64)+1)
+		}()
+	}
+	for range N {
+		<-ch
+	}
+	// Before refresh, we still have the initial value.
+	assert.Equal(t, float64(0), outerConfig.Get("n"))
+	err := outerConfig.GetStorage().Refresh(outerConfig, "n")
+	assert.NoError(t, err)
+	// After refresh, we get the sum from all the concurrent goroutines.
+	assert.Equal(t, float64(N), outerConfig.Get("n"))
+}
+
+func Test_JsonStorage_Locking_Interrupted(t *testing.T) {
+	outerConfig := NewFromFiles(TEST_FILENAME)
+	outerConfig.PersistInStorage("n")
+	outerConfig.Set("n", float64(0))
+	N := 100
+	ch := make(chan struct{}, N)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	for range N {
+		go func() {
+			defer func() { ch <- struct{}{} }()
+
+			config := NewFromFiles(TEST_FILENAME)
+			config.PersistInStorage("n")
+			storage := config.GetStorage()
+			assert.NotNil(t, storage)
+
+			err := storage.Lock(ctx, time.Millisecond)
+			if err != nil {
+				assert.ErrorIs(t, err, context.Canceled)
+				return
+			} else {
+				assert.NoError(t, err)
+			}
+			defer func() {
+				unlockErr := storage.Unlock()
+				assert.NoError(t, unlockErr)
+			}()
+
+			err = storage.Refresh(config, "n")
+			assert.NoError(t, err)
+			n := config.Get("n").(float64) //nolint:errcheck // let it panic in a test
```

Comment:
nitpick: Same as above, there is GetFloat64() which probably avoids the linter topic.